19.03.5.sh

```diff
@@ -511,6 +518,36 @@ do_install() {
 			)
 			exit 0
 			;;
+		sles)
+			(
+			set -x
+			$sh_c 'zypper refresh'
+			docker_pkg=$(zypper se -n -s docker | grep -E "\sdocker\s.*${docker_version//\./\\\.}.*$architecture" | sort --sort=v -k3 | tail -1 | cut -d '|' -f4 | cut -d ' ' -f2)
+			if [ -z "$docker_pkg" ]; then
+			    cat >&2 <<-'EOF'
+				No exact match found for SLES that matches docker \$docker_version for \$architecture.
+
+				Attempting to install the latest available version of docker \${docker_version%.*}.
```

Comment:
Was this a specific requirement for SLES? The script installs the requested version (`19.03.5` in this case) or fails. Getting a different version from the one you requested is unexpected.

Reply:
This should only affect the patch requested but will still fail if the major / minor version requested is incorrect i.e. since SUSE Linux only has 19.03.1 and 19.03.5, so if we ported these changes to the 19.03.4 or 19.03.6 (if/when it releases) install-docker script, the current implementation would still automatically pick up 19.03.5.

I do agree that getting a different version that then one requested is unexpected since, if a user did not have a preference for the patch they would like to use, they would have just decided to use the major.minor version of the script (i.e. 19.03), which is already symbolically linked to the latest version. However, in the case that we roll out a new version of install docker (i.e. 19.03.6) and the newest version does not have SLES support, then an existing user using SLES who has always requested for 19.03 (which now points to 19.03.6) would now suddenly be unable to install docker on their machines post the commit to point 19.03 -> 19.03.6 (since there is no support for 19.03.6 in SLES in this hypothetical situation).

Therefore, I think that either we can:
1) remove this logic and request users using SLES to point to a specific docker version / patch that is available on SLES rather than just 19.03, in which case they would be responsible for upgrading to the latest versions of docker on SLES when we add support.
2) retain this logic with the expectation that, for SLES, we perform a "best effort" installation to either get the patch requested or the latest patch adhering to that same major/minor version

@deniseschannon thoughts?

Reply:
discussed offline (described in comments)